### PR TITLE
docs: exclude some implementation details

### PR DESCRIPTION
**Pull request by coryan**
From time-to-time the template parameters of a function include implementation details to exclude the function from overload resolution, or to apply some template magic to deduce dependent types. These details do not need to be documented.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11645)

### CODE REVIEW

File: google/cloud/completion_queue.h [hunk 1]
```diff
@@ -146,6 +157,10 @@ class CompletionQueue {
    * by any thread blocked on this object's Run() member function. However, only
    * one callback in the handler is called at a time.
    *
+   * @deprecated Applications should have not need to call this function. The
```
Comment (devbww) [inline google/cloud/completion_queue.h:160]: s/not need/no need/

File: google/cloud/spanner/numeric.h [hunk 2]
```diff
@@ -257,7 +257,11 @@ StatusOr<Decimal<Mode>> MakeDecimal(double d) {
  */
 template <
     typename T, DecimalMode Mode,
-    typename std::enable_if<std::numeric_limits<T>::is_integer, int>::type = 0>
+    /// @cond implementation_details
+    typename std::enable_if<std::numeric_limits<T>::is_integer, int>::type = 0
+    /// @endcond
+
```
Comment (devbww) [inline google/cloud/spanner/numeric.h:263]: Extra blank line.

**Review by devbww — APPROVED** [review]

File: google/cloud/spanner/numeric.h (hunk 2 above)
Comment (coryan) [reply google/cloud/spanner/numeric.h]: Fixed

File: google/cloud/completion_queue.h (hunk 1 above)
Comment (coryan) [reply google/cloud/completion_queue.h]: Fixed

File: google/cloud/completion_queue.h [hunk 3]
```diff
@@ -110,27 +110,38 @@ class CompletionQueue {
   /**
    * Make an asynchronous unary RPC.
    *
+   * @deprecated Applications should have not need to call this function. The
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/completion_queue.h]: Fixed

**Pull request merged**